src_c/event.c

```diff
@@ -174,7 +174,34 @@ pg_event_filter(void *_, SDL_Event *event)
     else if (type == PGE_KEYREPEAT) {
         event->type = SDL_KEYDOWN;
     }
+    else if (type == SDL_MOUSEWHEEL) {
+        // Generate a MouseButtonDown event for compatibility.
+        // https://wiki.libsdl.org/SDL_MouseWheelEvent
+        SDL_Event newevent;
+        newevent.type = SDL_MOUSEBUTTONDOWN;
+        
+        int x, y;
+        VIDEO_INIT_CHECK();
+		
+        SDL_GetMouseState(&x, &y);
+        newevent.button.x = x;
+        newevent.button.y = y;
+        
+		
+        newevent.button.state = SDL_PRESSED;
+        newevent.button.clicks = 1;
+        
+        if (event->wheel.y != 0) {
+            newevent.button.button = (event->wheel.y > 0) ? 4 : 5;
+        }
+        else if (event->wheel.x != 0) {
+            newevent.button.button = (event->wheel.x > 0) ? 4 : 5;
+        }
```

Comment:
I get a bunch of mouse button events with button == 80 when I scroll. Apparently pos can be (0, 0), at least on my touchpad. So it shouldn't generate a mouse button event in that case.

Reply:
> I wonder if this might a better idea for the button events: Add 2 if button id >= 4 (and < 96)? > So the buttons become 6, 7... instead of 4, 5...
> SDL_BUTTON_LEFT
> SDL_BUTTON_MIDDLE
> SDL_BUTTON_RIGHT
> SDL_BUTTON_X1
> SDL_BUTTON_X2
> I suggest we add these constants and also add constants for BUTTON_WHEELUP and BUTTON_WHEELDOWN (also for SDL 1).

Using 6/7 instead of 4/5 still break compatibility, but that's needed I think.

BUTTON_WHEELUP / BUTTON_WHEELDOWN should be 4/5 in SDL1 and 6/7 in SDL2 in this way,

but some apps are still using if event.button == 4 or 5, so taking changes in old apps are necessary.

Reply:
> I get a bunch of mouse button events with button == 80 when I scroll. Apparently pos can be (0, 0), at least on my touchpad. So it shouldn't generate a mouse button event in that case.

if it generates button == 80 and button == 4/5 at the same time, 

then the button==80 event can be safely filtered I think?